test/functional/cursor.test.js

```diff
@@ -4312,4 +4305,64 @@ describe('Cursor', function () {
       });
     });
   });
+
+  context('sort', function () {
+    const findSort = (input, output) =>
+      withMonitoredClient('find', function (client, events, done) {
+        const db = client.db('test');
+        db.collection('test_sort_dos', (err, collection) => {
+          expect(err).to.not.exist;
+          const cursor = collection.find({}, { sort: input });
+          expect(cursor.sortValue).to.deep.equal(output);
+          cursor.next(err => {
+            expect(err).to.not.exist;
+            expect(events[0].command.sort).to.deep.equal(output);
+            cursor.close(done);
+          });
+        });
+      });
+
+    const cursorSort = (input, output) =>
+      withMonitoredClient('find', function (client, events, done) {
+        const db = client.db('test');
+        db.collection('test_sort_dos', (err, collection) => {
+          expect(err).to.not.exist;
+          const cursor = collection.find({}).sort(input);
+          expect(cursor.sortValue).to.deep.equal(output);
+          cursor.next(err => {
+            expect(err).to.not.exist;
+            expect(events[0].command.sort).to.deep.equal(output);
+            cursor.close(done);
+          });
+        });
+      });
+
+    it('should use find options object', findSort({ alpha: 1 }, { alpha: 1 }));
+    it('should use find options string', findSort('alpha', { alpha: 1 }));
+    it('should use find options shallow array', findSort(['alpha', 1], { alpha: 1 }));
+    it('should use find options deep array', findSort([['alpha', 1]], { alpha: 1 }));
+
+    it('should use cursor.sort object', cursorSort({ alpha: 1 }, { alpha: 1 }));
+    it('should use cursor.sort string', cursorSort('alpha', { alpha: 1 }));
+    it('should use cursor.sort shallow array', cursorSort(['alpha', 1], { alpha: 1 }));
+    it('should use cursor.sort deep array', cursorSort([['alpha', 1]], { alpha: 1 }));
+
+    it('formatSort', () => {
```

Comment:
Should there be some more tests that try multiple sort keys?
Also repeated keys in the list of tuples form will implicitly be reduced to one since we use an object, let's just add a case to make sure its as expected.

Reply:
added some multi tests